zio-http/src/test/scala/zhttp/middleware/HttpMiddlewareSpec.scala

```diff
@@ -14,6 +14,14 @@ object HttpMiddlewareSpec extends DefaultRunnableSpec {
     UIO(Response.ok).delay(1 second)
   }
 
+  val app1: HttpApp[Any with Clock, Nothing] = HttpApp.collectM { case Method.GET -> !! / "create" =>
+    UIO(Response.status(Status.CREATED)).delay(1 second)
+  }
+
+  val app2: HttpApp[Any with Clock, Nothing] = HttpApp.collectM { case Method.GET -> !! / "bad_request" =>
+    UIO(Response.status(Status.BAD_REQUEST)).delay(1 second)
+  }
```

Comment:
We can use simpler apps for testing such as 

```scala
HttpApp.ok.addHeader("name", "left")
```

Then assert using "hasHeader"